juriscraper/opinions/united_states/state/nd.py

```diff
@@ -50,11 +54,30 @@ def _process_html(self) -> None:
 
             for idx, txt in enumerate(raw_values[:5]):
                 if idx == 0:
-                    txt, _ = self.clean_name(txt)
+                    # Separate case name and citation if present
+                    match = re.match(
+                        r"^(.*?)(\s*((\d{4}\sND\s\d+)|(1 \d\.N\.W\d d\d+)))?\s*$",
+                        txt,
+                    )
+                    if match:
+                        case_name = match.group(1).strip()
+                        # If matched with the second regex (1 \d\.N\.W\d d\d+), set citation to ""
+                        if match.group(5):
+                            citation = ""
+                        else:
+                            citation = (
+                                match.group(2).strip()
+                                if match.group(2)
+                                else ""
+                            )
+                        txt = case_name
+                    else:
+                        citation = ""
+                    txt = self.clean_name(txt)
+                    values.append(citation)
```

Comment:
this is getting a bit unwieldy.  I think the few examples of nw3d with broken citations should be ignored.  focus solely on the ND format and ignore the rest.  It's not worth the hassle in my opinion for bad data.